lib/web_ui/lib/src/engine/text/layout_service.dart

```diff
@@ -178,6 +175,13 @@ class TextLayoutService {
             currentLine = currentLine.nextLine();
           } else {
             // Normal line break.
+            currentLine.revertToLastBreakOpportunity();
+            // If a revert had occurred in the line, we need to revert the span
+            // index accordingly.
```

Comment:
Mention the special case that when no revert happened then `revertedToSpan` will simply be equal `span`.